src/thumbnailer.rs

```diff
@@ -195,87 +201,91 @@ impl Thumbnailer {
     }
 
     fn save_thumbnail_to_temp(thumbnailer: Thumbnailer) -> Result<Thumbnailer, String> {
-        &thumbnailer
-            .thumbnail
-            .as_ref()
-            .unwrap()
-            .save_with_format(
-                thumbnailer.temp_path.to_str().unwrap(),
-                image::ImageFormat::PNG,
-            )
-            .map_err(|e| {
-                error!("{}", e);
-                "Failed to save thumbnail".to_owned()
-            });
-        Ok(thumbnailer)
-    }
-
-    fn update_metadata(thumbnailer: Thumbnailer) -> Result<Thumbnailer, String> {
-        let mut chunks = {
-            let mut input = std::fs::File::open(&thumbnailer.temp_path)
-                .map_err(|_e| "Failed to open thumbnailer in temporary dir".to_owned())?;
-            crate::png::Png::decode(&mut input).map_err(|_e| "Failed decoding chunks".to_owned())?
+        let temp_path = format!("{}.tmp", thumbnailer.destination_path.to_str().unwrap());
+        let output = std::fs::OpenOptions::new()
+            .write(true)
+            .truncate(true)
+            .create(true)
+            .open(&temp_path)
+            .map_err(|e| format!("Failed to open thumbnailer in temporary dir: {}", e))?;
+
+        let thumbnail = thumbnailer.thumbnail.as_ref().unwrap();
+        let (ct, bits) = match thumbnail.color() {
+            image::ColorType::L8 => (png::ColorType::Grayscale, png::BitDepth::Eight),
+            image::ColorType::L16 => (png::ColorType::Grayscale, png::BitDepth::Sixteen),
+            image::ColorType::La8 => (png::ColorType::GrayscaleAlpha, png::BitDepth::Eight),
+            image::ColorType::La16 => (png::ColorType::GrayscaleAlpha, png::BitDepth::Sixteen),
+            image::ColorType::Rgb8 => (png::ColorType::RGB, png::BitDepth::Eight),
+            image::ColorType::Rgb16 => (png::ColorType::RGB, png::BitDepth::Sixteen),
+            image::ColorType::Rgba8 => (png::ColorType::RGBA, png::BitDepth::Eight),
+            image::ColorType::Rgba16 => (png::ColorType::RGBA, png::BitDepth::Sixteen),
+            _ => return Err("unsupported format".to_string()),
         };
+        let mut encoder = png::Encoder::new(output, thumbnail.width(), thumbnail.height());
+        encoder.set_color(ct);
+        encoder.set_depth(bits);
+        let mut writer = encoder
+            .write_header()
+            .map_err(|e| format!("Error writing PNG header: {}", e))?;
 
         let uri_raw = Thumbnailer::calculate_path_uri(
             thumbnailer.use_full_path_for_md5,
             &thumbnailer.source_path,
         );
-        let uri = crate::png::Chunk::new_text("Thumb::URI", uri_raw).unwrap();
-        chunks.insert(1, uri);
+        writer
+            .write_chunk(PNG_TEXT_KIND, &text_chunk("Thumb::URI", uri_raw).unwrap())
```

Comment:
Curious: were all these chunk related functions available in previous versions of the png library? Is this the reason for the upgrade?

Reply:
Tbh, the dep updates were just done as part of the png create research/development. I had initially removed the png dependency (it wasn't used at all), and then when I ended up needing it I added the latest version.

Want me to remove or isolate the version bumps? I'm fine either way.